Docs fix metric storage (#13327)
netdata/netdata: database/README.md

   details see [here](/database/engine/README.md).

2. `ram`, data are purely in memory. Data are never saved on disk. This mode uses `mmap()` and supports [KSM](#ksm).

3. `save`, data are only in RAM while Netdata runs and are saved to / loaded from disk on Netdata restart. It also
   uses `mmap()` and supports [KSM](#ksm).

4. `map`, data are in memory mapped files. This works like the swap. When Netdata writes data on its memory, the Linux
   kernel marks the related memory pages as dirty and automatically starts updating them on disk. Unfortunately we
   cannot control how frequently this works. The Linux kernel uses exactly the same algorithm it uses for its swap
   memory. This mode uses `mmap()` but does not support [KSM](#ksm). _Keep in mind though, this option will have a
   constant write on your disk._

5. `alloc`, like `ram` but it uses `calloc()` and does not support [KSM](#ksm). This mode is the fallback for all others
   except `none`.

6. `none`, without a database (collected metrics can only be streamed to another Netdata).

## Which database mode to use

> Netdata Agents support multiple database modes.

Reviewer: Line not needed, the section above listed them.

The default mode `[db].mode = dbengine` has been designed to scale for longer retentions.

The other available database modes are designed to minimize resource utilization and should usually be considered on **

> parent - child** setups at the children side.

Reviewer: "at the children side, and only when the resource constraints are very strict"

So,

> - On a single node setup, use `[db].mode = dbengine` to increase retention.

Reviewer: remove "to increase retention".

- On a [parent - child](/docs/metrics-storage-management/how-streaming-works) setup, use `[db].mode = dbengine` on the
  parent to increase retention and a more resource efficient mode (like `save`, `ram` or `none`) for the child to

>   minimize resources utilization.

Reviewer: "resource utilization"

## Choose your database mode

You can select the database mode by editing `netdata.conf` and setting:

```conf
[db]
  # dbengine (default), ram, save (the default if dbengine not available), map (swap like), none, alloc
  mode = dbengine
```

## Netdata Longer Metrics Retention

> Metrics retention affects 3 parameters on the operation of a Netdata Agent:

Reviewer: "Metrics retention is controlled by 3 configuration parameters"

1. The disk space required to store the metrics.
2. The memory the Netdata Agent will require to have that retention available for queries.
3. The CPU resources that will be required to query longer time-frames.

As retention increases, the resources required to support that retention increase too.

> Since Netdata Agents usually run on the edge, inside production systems, Netdata Agent **parents** should be considered.

Reviewer: "on production systems", not "inside"

When having a **parent - child** setup, the child (the Netdata Agent running on a production system) delegates all its
functions, including longer metrics retention and querying, to the parent node that can dedicate more resources to this
task. A single Netdata Agent parent can centralize multiple children Netdata Agents (dozens, hundreds, or even thousands
depending on its available resources).

> ## Running Netdata in embedded devices

Reviewer: "on" embedded devices

> ## Running Netdata in embedded devices

Reviewer: This whole section is problematic. We're not even saying here which memory mode they need to run with. 
I suggest we move this to the "high performance netdata" and suggest the following:

- If your devices can stream data to a parent without incurring significant traffic costs, use memory mode RAM and set up streaming
- If cost of data transfer is a concern, use memory mode save or map (to tell you the truth I never understood myself which is appropriate for which use case). 

You'll find that we are already suggesting lower retention in the high performance netdata doc.

Embedded devices usually have very limited RAM resources available.

There are 2 settings for you to tweak:

1. `[db].update every`, which controls the data collection frequency
2. `[db].retention`, which controls the size of the database in memory (except for `[db].mode = dbengine`)

By default `[db].update every = 1` and `[db].retention = 3600`. This gives you an hour of data with per second updates.

If you set `[db].update every = 2` and `[db].retention = 1800`, you will still have an hour of data, but collected once
every 2 seconds. This will **cut in half** both CPU and RAM resources consumed by Netdata. Of course experiment a bit.
On very weak devices you might have to use `[db].update every = 5` and `[db].retention = 720` (still 1 hour of data, but
1/5 of the CPU and RAM resources).

You can also disable [data collection plugins](/collectors/README.md) you don't need. Disabling such plugins will also
free both CPU and RAM resources.

> ## Running a dedicated parent Netdata server

Reviewer: Not sure why need a section on this in this document. References in the previous mentions of parents or streaming will be sufficient.

Author: I will omit it

Netdata allows streaming data between Netdata nodes in real-time. This allows you having one or more parent Netdata
servers that will maintain the entire database for all the nodes that connect to them (their children), and will also
run health checks/alarms for all these nodes. Explore this feature following the following
section: [enable streaming between nodes](/docs/metrics-storage-management/enable-streaming.md)

> ## KSM

Reviewer: See https://github.com/rothgar/netdata-wiki/blob/master/Memory-Deduplication---Kernel-Same-Page-Merging---KSM.md and bring in the first few lines in again. We lost some valuable info from omitting those two lines.

   cannot control how frequently this works. The Linux kernel uses exactly the same algorithm it uses for its swap
   memory. This mode uses `mmap()` but does not support [KSM](#ksm). _Keep in mind though, this option will have a
   constant write on your disk._

5. `alloc`, like `ram` but it uses `calloc()` and does not support [KSM](#ksm). This mode is the fallback for all others
   except `none`.

6. `none`, without a database (collected metrics can only be streamed to another Netdata).

## Which database mode to use

The default mode `[db].mode = dbengine` has been designed to scale for longer retentions and is the only mode suitable
for parent Agents in the _Parent - Child_ setups

The other available database modes are designed to minimize resource utilization and should only be considered on
[Parent - Child](/docs/metrics-storage-management/how-streaming-works) setups at the children side and only when the
resource constraints are very strict.

So,

- On a single node setup, use `[db].mode = dbengine`.
- On a [Parent - Child](/docs/metrics-storage-management/how-streaming-works) setup, use `[db].mode = dbengine` on the

>   parent to increase retention and a more resource efficient modes like, `dbengine` with light retention settings,

Reviewer 2: ```suggestion
  parent to increase retention, a more resource efficient mode like, `dbengine` with light retention settings, and
```

  `save`, `ram` or `none` modes for the children to minimize resource utilization.

## Choose your database mode

You can select the database mode by editing `netdata.conf` and setting:

```conf
[db]
  # dbengine (default), ram, save (the default if dbengine not available), map (swap like), none, alloc
  mode = dbengine
```

## Netdata Longer Metrics Retention

Metrics retention is controlled only by the disk space allocated to storing metrics. But it also affects the memory and
CPU required by the agent to query longer timeframes.

Since Netdata Agents usually run on the edge, on production systems, Netdata Agent **parents** should be considered.
When having a [**parent - child**](/docs/metrics-storage-management/how-streaming-works.md) setup, the child (the
Netdata Agent running on a production system) delegates all its functions, including longer metrics retention and
querying, to the parent node that can dedicate more resources to this task. A single Netdata Agent parent can centralize
multiple children Netdata Agents (dozens, hundreds, or even thousands depending on its available resources).

## Running Netdata on embedded devices

> Embedded devices usually have very limited RAM resources available.

Reviewer 2: ```suggestion
Embedded devices typically have very limited RAM resources available.
```

> There are 2 settings for you to tweak:

Reviewer 2: ```suggestion
There are two settings for you to configure:
```
